### PR TITLE
Hour angle, local sidereal time convenience methods

### PR DESCRIPTION
**Pull request by bmorris3**
This makes convenience methods for two quantities that show up on all TUI displays and should be implemented in astroplan. They are: 

* `Observer.local_sidereal_time(time, [...])`
* `Observer.target_hour_angle(time, target)`

These features were requested in #88.

cc @ejeschke @wtgee

### CODE REVIEW

File: astroplan/tests/test_observer.py [hunk 1]
```diff
@@ -1156,3 +1157,33 @@ def vectorize_timing(n_targets):
...
+
+def test_local_sidereal_time():
+    time = Time('2005-02-03 00:00:00')
+    location = EarthLocation.from_geodetic(10*u.deg, 40*u.deg, 0*u.m)
+    obs = Observer(location=location)
+    # test sidereal time
+    astroplan_lst = obs.local_sidereal_time(time)
+    # Compute this with print_pyephem_lst()
+    pyephem_lst = 2.5005375428099104*u.rad
+    assert_quantity_allclose(pyephem_lst, astroplan_lst, atol=0.01*u.deg)
```
Comment (cdeil) [inline astroplan/tests/test_observer.py]:
```
Signature: assert_quantity_allclose(actual, desired, rtol=1e-07, atol=None, **kwargs)
```
So I'd say here, `actual=astroplan_lst` and `expected=pyephem_lst`, i.e. exchange the order of the arguments
to get a clearer message should it fail somewhere.

(Maybe have a look around checking that this is done correctly throughout astroplan?)

File: astroplan/observer.py [hunk 2]
```diff
@@ -1620,3 +1620,72 @@ def is_night(self, time, horizon=0*u.deg, obswl=None):
...
+            Passed to the ``kind`` argument of
+            `~astropy.time.Time.sidereal_time`
+
+        model : str or `None`; optional
+            Passed to the ``model`` argument of
+            `~astropy.time.Time.sidereal_time`
+
+        Returns
+        -------
+
```
Comment (cdeil) [inline astroplan/observer.py]: Add what's returned here?
(or remove the `Returns` section if it's obvious)

**Comment by bmorris3** [comment]
Thanks @cdeil, corrections made.

File: astroplan/observer.py [hunk 3]
```diff
@@ -1620,3 +1620,73 @@ def is_night(self, time, horizon=0*u.deg, obswl=None):
...
+            the `~astropy.time.Time` initializer, so it can be anything that
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object)
+
+        kind : {'mean', 'apparent'} (optional)
+            Passed to the ``kind`` argument of
+            `~astropy.time.Time.sidereal_time`
+
+        model : str or `None`; optional
+            Passed to the ``model`` argument of
```
Comment (eteq) [inline astroplan/observer.py]: Maybe just add something like ``The precession/nutation model to assume - see `~astropy.time.Time.sidereal_time` for more details.`` ?  That way the user gets at least some hint of what sort of "model" it is.

File: astroplan/observer.py [hunk 4]
```diff
@@ -1620,3 +1620,73 @@ def is_night(self, time, horizon=0*u.deg, obswl=None):
...
+            `~astropy.time.Time` will accept (including a `~astropy.time.Time`
+            object)
+
+        target : coordinate object (i.e. `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`) or list
+            Target celestial object(s)
+
+        Returns
+        -------
+        hour_angle : `~astropy.coordinates.Angle`
+            The hour angle of the target at ``time``
```
Comment (eteq) [inline astroplan/observer.py]: Might be worth saying ``hour angle(s)`` to make it clear this could be a vector `Angle`?

**Comment by bmorris3** [comment]
Addressed comments by @eteq, merging!

**Pull request merged**